src/platforms/web/runtime/components/transition.js

```diff
@@ -160,6 +160,7 @@ export default {
     if (
       oldChild &&
       oldChild.data &&
+      oldChild.elm.nodeType !== 8 && // Node.COMMENT_NODE
```

Comment:
`oldChild` could be `undefined`, so check existence should be necessary.
```js
oldChild.elm && oldChild.elm.nodeType !== 8
```

Reply:
In which situation would this happen?

Reply:
This is why 'async components with transition-mode out-in' test failing. 🙂